src/test/java/org/zalando/nakadi/controller/CursorsControllerTest.java

```diff
@@ -60,27 +71,40 @@ public CursorsControllerTest() throws Exception {
         featureToggleService = mock(FeatureToggleService.class);
         when(featureToggleService.isFeatureEnabled(any())).thenReturn(true);
 
-        final CursorsController controller = new CursorsController(cursorsService, featureToggleService);
+        final SubscriptionDbRepository subscriptionRepository = mock(SubscriptionDbRepository.class);
+        final EventTypeRepository eventTypeRepository = mock(EventTypeRepository.class);
+        doReturn(buildDefaultEventType()).when(eventTypeRepository).findByName(any());
+        doReturn(RandomSubscriptionBuilder.builder().build()).when(subscriptionRepository).getSubscription(any());
+        final CursorsController controller = new CursorsController(cursorsService, featureToggleService,
+                subscriptionRepository, eventTypeRepository);
         final MappingJackson2HttpMessageConverter jackson2HttpMessageConverter =
                 new MappingJackson2HttpMessageConverter(objectMapper);
 
+        final SecuritySettings settings = mock(SecuritySettings.class);
+        doReturn(SecuritySettings.AuthMode.OFF).when(settings).getAuthMode();
+        doReturn("nakadi").when(settings).getAdminClientId();
+        doReturn(true).when(featureToggleService).isFeatureEnabled(CHECK_PARTITIONS_KEYS);
+
         mockMvc = standaloneSetup(controller)
                 .setMessageConverters(new StringHttpMessageConverter(), jackson2HttpMessageConverter)
+                .setCustomArgumentResolvers(new ClientResolver(settings, featureToggleService))
                 .build();
     }
 
     @Test
-    public void whenCommitValidCursorsThenOk() throws Exception {
-        when(cursorsService.commitCursors(any(), any())).thenReturn(true);
+    public void whenCommitValidCursorsThenNoContent() throws Exception {
+        when(cursorsService.commitCursors(any(), any()))
+                .thenReturn(new HashMap<>());
         putCursors(DUMMY_CURSORS)
-                .andExpect(status().isOk());
+                .andExpect(status().isNoContent());
     }
 
     @Test
-    public void whenCommitOldCursorsThenNoContent() throws Exception {
-        when(cursorsService.commitCursors(any(), any())).thenReturn(false);
+    public void whenCommitInvalidCursorsThenOk() throws Exception {
+        when(cursorsService.commitCursors(any(), any()))
+                .thenReturn(Collections.singletonMap(any(), false));
         putCursors(DUMMY_CURSORS)
-                .andExpect(status().isNoContent());
+                .andExpect(status().isOk());
```

Comment:
I think it would be interesting to test not only the status code but also the body, as it should contain the list of commit results.